app/src/androidTest/java/com/example/avjindersinghsekhon/minimaltodo/testApp/tests/tests/DemoTest.java

```diff
@@ -0,0 +1,156 @@
+
+        onView(withId(R.id.toDoListItemTextview)).perform(swipeLeft());
+
+    }
+
+    @Test
+    public void TestCase1(){
+
+        //Verify "Minimal" and "You don't have any todos" text is displayed
+        onView(withText(R.string.app_name)).check(matches(isDisplayed()));
+        onView(withText(R.string.no_to_dos)).check(matches(isDisplayed()));
+
+        //Verify Image is displayed above "You don't have any todos" text
+        onView(allOf(withParent(withId(R.id.toDoEmptyView)), Matchers.<View>instanceOf(ImageView.class))).check(isAbove(withText(R.string.no_to_dos)));
+
+        //Verify "Minimal" text is part of the toolbar
+        onView(allOf(withText(R.string.app_name), withParent(withId(R.id.toolbar)))).check(matches(isCompletelyDisplayed()));
+
+        //Verify Add button exist
+        onView(allOf(withId(R.id.addToDoItemFAB), isCompletelyDisplayed())).check(matches(isClickable()));
+
+        //Verify button with content-desc "More options" exist, has a parent with a brother with "Minimal" text and is part of toolbar
+        onView(allOf(withContentDescription("More options"), withParent(hasSibling(withText(R.string.app_name))), isDescendantOfA(withId(R.id.toolbar)))).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void TestCase2(){
+
+        //Click the add button and verify x is displayed
+        onView(withId(R.id.addToDoItemFAB)).perform(click());
+        onView(allOf(withParent(withId(R.id.toolbar)), withContentDescription("Navigate up"))).check(matches(isDisplayed()));
+
+        //Click x and verify "You don't have any todos" is displayed
+        onView(allOf(withParent(withId(R.id.toolbar)), withContentDescription("Navigate up"))).perform(click());
+        onView(withText(R.string.no_to_dos)).check(matches(isDisplayed()));
+        closeSoftKeyboard();
+
+        //Click the add button
+        onView(withId(R.id.addToDoItemFAB)).perform(click());
+
+        //Close keyboard. Verify "edit text" is displayed and it has a parent. Verify there is a "Remind me" text positioned in between 2 buttons
+        closeSoftKeyboard();
+        onView(allOf(withId(R.id.userToDoEditText), withParent(withId(R.id.toDoCustomTextInput)))).check(matches(isDisplayed()));
+        onView(withText(R.string.remind_me)).check(isBelow(withId(R.id.makeToDoFloatingActionButton))).check(isLeftOf(withId(R.id.toDoHasDateSwitchCompat)));
+
+        //Add a to-do with "MY TO-DO" text and verify the added text is displayed
+        onView(withId(R.id.userToDoEditText)).perform(typeText("MY TODO"));
+        onView(withText("MY TODO")).check(matches(isDisplayed()));
+
+        //Turn on toggle and verify text starting with "Reminder set" is displayed under another "Today" text
+        onView(withId(R.id.toDoHasDateSwitchCompat)).perform(click());
+        onView(withText(startsWith("Reminder set"))).check(isBelow(withText(R.string.date_reminder_default)));
```

Comment:
This method is deprecated.
Use isCompletelyBelow(Matcher) instead.

https://developer.android.com/reference/android/support/test/espresso/assertion/PositionAssertions.html#isCompletelyBelow(org.hamcrest.Matcher%3Candroid.view.View%3E)